### PR TITLE
Base on alpine and support more parameters

### PR DESCRIPTION
**Pull request by so0k**
Changes:

- Use alpine as base
- Add dockerignore and make scripts
- Add Env var to specify analytics range as this depends on CF Plan
- Add Kubernetes deployment (will move to Helm chart soon)

### CODE REVIEW

**Comment by so0k** [comment]
I'm wondering if it wouldn't be better to use [the python-cloudfare package](https://github.com/cloudflare/python-cloudflare).

```bash
pip install python-cloudflare
```

Export env vars
```bash
export CF_API_EMAIL=
export CF_API_KEY=
```

Sample python script

```python
from CloudFlare import CloudFlare
from pprint import pprint

cf = CloudFlare()
zone = cf.zones.get(params={'name':'yourdomain.com'})
dashboard=cf.zones.analytics.dashboard.get(zone[0]['id'],params={'since':'-360'})

len(dashboard['timeseries'])
pprint(dashboard['timeseries'][0])
```

and make some changes to the conversion of the cloudflare response to the datadog timeseries... (I think the only difference is that the requests object has an extra layer to unpack)

The advantages are:

- script no longer needs to map keys from env var (the python-cloudflare package takes care of that)
- no need to figure out the zone id in advance

**Comment by jgmize** [comment]
Thanks @so0k, I haven't had a chance to give this a thorough review yet, but this is looking great to me so far! I'm also really happy to hear that you're planning on making a helm chart later; I've filed #4 to track that.

@glogiotatidis, if you have time to review this in the next few business days please take a look; I'd love to get this moved over from whichever of our legacy clusters this is currently running on to one of our k8s clusters.

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,11 @@
-FROM python:2-onbuild
+FROM python:2-alpine
+
+RUN apk add --no-cache tzdata \
+  && ln -sf /usr/share/zoneinfo/Asia/Singapore /etc/localtime
```
Comment (glogiotatidis) [inline Dockerfile]: * Why is this needed? 
 * UTC is a more sane default. 
 * We can move the liking part to an `entrypoint` script so it's easy to select timezone via env variables.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,71 @@
+SHELL=/bin/bash
+DIR_NAME ?= $(notdir $(shell pwd))
+
+DOCKER_REGISTRY ?= registry.honestbee.com/
```
Comment (glogiotatidis) [inline Makefile]: I'd prefer to use docker registry as  the default public registry

File: README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Cloudflare to Datadog Bridge #
 
+[![Docker Repository on Quay](https://quay.io/repository/honestbee/cloudflare-datadog/status "Docker Repository on Quay")](https://quay.io/repository/honestbee/cloudflare-datadog)
```
Comment (glogiotatidis) [inline README.md]: Let's change this to https://hub.docker.com/r/mozmeao/cloudflare-datadog

File: Dockerfile [hunk 4]
```diff
@@ -1,3 +1,11 @@
-FROM python:2-onbuild
+FROM python:2-alpine
+
+RUN apk add --no-cache tzdata \
```
Comment (glogiotatidis) [inline Dockerfile]: 👍 for `apt --no-cache`

File: k8s-bundle.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: dd-cf-agent
+    spec:
+      containers:
+      - name: dd-cf-agent
+        image: registry.honestbee.com/honestbee/cloudflare-datadog:latest
```
Comment (glogiotatidis) [inline k8s-bundle.yaml]: Update to point to https://hub.docker.com/r/mozmeao/cloudflare-datadog please

File: k8s-bundle.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+              key: zone
+        - name: "SINCE"
+          valueFrom: 
+            secretKeyRef:
+              name: dd-cf-agent-secret
+              key: since
+        resources: {}
+        terminationMessagePath: /dev/termination-log
+      imagePullSecrets: 
+      - name: honestbee-registry
```
Comment (glogiotatidis) [inline k8s-bundle.yaml]: And remove this.

File: env2secret.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python
```
Comment (glogiotatidis) [inline env2secret.py]: Do you mind getting this file pep8 compatible to keep things tidy?

**Review by glogiotatidis — CHANGES_REQUESTED** [review]
Hi @so0k thanks for the suggestions, additions and the PR! 🍻 

Since this is coming directly from honestbee/master is tuned for your needs. Could you please replacing the custom parts / variables with more generic ones and cleaning it up a bit?

File: README.md [hunk 8]
```diff
@@ -42,4 +75,48 @@ Run:
...
+
+Push to Registry (update Makefile with registry server applicable)
+```
+make push
+```
+
+Convert environment file to kubernetes secret and create:
+```
+./env2secret.py -s dd-cf-agent-secret
+kubectl apply -f secret.yaml
```
Comment (bookshelfdave) [inline README.md]: did we decide upon a namespace to run this in?

File: Makefile (hunk 2 above)
Comment (so0k) [reply Makefile]: I forgot that was in there :) I think the Makefile can be removed all together and just use an automated repository for docker builds and no other automation (no k8s manifest also)

File: Dockerfile (hunk 1 above)
Comment (so0k) [reply Dockerfile]: Ok, adding env var support for this (entrypoints are the new init scripts)

File: k8s-bundle.yaml (hunk 5 above)
Comment (so0k) [reply k8s-bundle.yaml]: I think we should drop the k8s-bundle.yaml and refer to the helm chart? (Unless we want to support ppl who don't want to install helm - but different ppl may have different deployment methods)

File: env2secret.py (hunk 7 above)
Comment (so0k) [reply env2secret.py]: I thought about dropping the Makefile and the need for this file as Helm is much better at handling release on k8s

**Comment by so0k** [comment]
With the support for Timezone, test as follows:

```shell
$ docker run -it --rm cloudflare-datadog date
Tue Mar 14 16:55:54 UTC 2017
$ docker run -it --rm -e "TIMEZONE=Asia/Singapore" cloudflare-datadog date
Wed Mar 15 00:56:08 SGT 2017
```

List available timezone info:
```shell
$ docker run -it --rm cloudflare-datadog sh -c "apk add --no-cache tree && tree /usr/share/zoneinfo/"
/usr/share/zoneinfo/
├── Africa
│   ├── Abidjan
│   ├── Accra
│   ├── Addis_Ababa
│   ├──...
```

File: env2secret.py (hunk 7 above)
Comment (glogiotatidis) [reply env2secret.py]: I agree on dropping the makefile and keeping only Helm. Both our teams are using Helm so lets keep this simple.

**Comment by glogiotatidis** [comment]
I linked this repository to docker hub as automated build thus we can drop the Makefile and related files.

**Comment by so0k** [comment]
Makefile can still be handy to test a build locally, I removed all private registry stuff.

**Comment by glogiotatidis** [comment]
LGTM! I'll squash merge this and work on #5. Thanks again @so0k ⭐️

**Review by glogiotatidis — APPROVED** [review]

**Pull request merged**